rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -885,8 +892,17 @@ public InputStream jobFullLogs(String sessionId, String jobId, String session) t
                 }
             }
 
-            // will produce HTTP 204 code if null
-            return streams.isEmpty() ? null : new SequenceInputStream(Collections.enumeration(streams));
+            builder = Response.ok()
+                              .entity(streams.isEmpty() ? IOUtils.toInputStream("",
+                                                                                Charset.forName(PASchedulerProperties.FILE_ENCODING.getValueAsString()))
+                                                        : new SequenceInputStream(Collections.enumeration(streams)));
+
+            if (DESTINATION_FILE.equals(destination)) {
+                builder = builder.header(HttpHeaders.CONTENT_DISPOSITION,
+                                         "attachment; filename=job_" + jobId + "_logs.txt");
```

Comment:
shouldn't the extension be .log for more convenience and easy opening with log monitoring software ?

Reply:
ok